lkcrypto.c

```diff
@@ -198,18 +272,115 @@ static int decrypt_encrypt(lua_State *L, int dec_enc)
 		memcpy(iv, iv_orig, iv_len);
 	}
 	buffer = kmalloc(ciphertext_len, GFP_ATOMIC);
+	if (buffer == NULL) {
+		return luaL_error(L, "buffer allocation failed");
+	}
 	memcpy(buffer, ciphertext, ciphertext_len);
 	result = enc_dec(tfm, key, key_len, iv, iv_len, buffer, ciphertext_len,
-			 dec_enc);
+			 operation);
 
 	if (!result) {
 		lua_pushlstring(L, buffer, ciphertext_len);
 		kfree(buffer);
 		return 1;
 	} else {
 		kfree(buffer);
-		return luaL_error(L, "%s failed %d",
-				  dec_enc ? "encrypt" : "decrypt");
+		return luaL_error(L, "%s failed ",
```

Comment:
nit: trailing space

Reply:
Addressed

Reply:
scatterlist is ok to be allocated on stack, the problem mentioned is happening when you point scatterlist (`sg_set_buf`) to buffer that is allocated on stack (i.e. not scatterlist itself)